### PR TITLE
Unset `_id: null` to let it be autogenerated

### PR DESCRIPTION
**Pull request by GromNaN**
Tentative fix for #2945

### Checklist

- [x] Add tests and ensure they pass
- [x] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: src/Eloquent/Builder.php [hunk 1]
```diff
@@ -233,8 +234,8 @@ public function createOrFirst(array $attributes = [], array $values = []): Model
             try {
                 $document = $collection->findOneAndUpdate(
                     $attributes,
-                    // Before MongoDB 5.0, $setOnInsert requires a non-empty document.
-                    // This should not be an issue as $values includes the query filter.
+                    // Before MongoDB 5.0, $setOnInsert requires a non-empty document,
+                    // this should not be an issue as $values include the query filter.
```
Comment (jmikola) [inline src/Eloquent/Builder.php]: Grammar was correct as-is. `$values` is a singular variable, so the verb would be plural.

File: tests/ModelTest.php [hunk 2]
```diff
@@ -1151,4 +1151,21 @@ public function testUpdateOrCreate(array $criteria)
...
+    {
+        $user = User::create(['_id' => null, 'email' => 'foo@bar']);
+        $this->assertNotNull(ObjectId::class, $user->id);
+        $this->assertSame(1, User::count());
+    }
+
+    public function testUpdateOrCreateWithNullId()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('You must provide attributes to check for duplicates');
```
Comment (jmikola) [inline tests/ModelTest.php:1165]: Noted that `expectExceptionMessage()` is a substring match.

File: tests/ModelTest.php [hunk 3]
```diff
@@ -1151,4 +1151,21 @@ public function testUpdateOrCreate(array $criteria)
...
+        User::updateOrCreate(
+            ['_id' => null],
```
Comment (jmikola) [inline tests/ModelTest.php:1167]: Do users typically pass an explicit `null` value to this method, or would this argument come from another source such as an empty model object?

Comment (GromNaN) [reply tests/ModelTest.php:1167]: From the [original issue](https://github.com/mongodb/laravel-mongodb/issues/2945), `null` is a fallback value.

```php
$id = $request->get('id') ?? null;
Model::updateOrCreate(['_id' => $id], $data);
```

**Review by jmikola — APPROVED** [review]

**Pull request merged**